src/lib/foreign-field.ts

```diff
@@ -661,7 +696,7 @@ type Constructor<T> = new (...args: any[]) => T;
 
 function provable<F extends ForeignField>(
   Class: Constructor<F> & { check(x: ForeignField): void }
-): ProvablePure<F> {
+): ProvablePureExtended<F, string> {
```

Comment:
making this support `toJSON` etc. makes a nicer inferred `Provable` for `ForeignCurve` and `EcdsaSignature` as well